design/conditional-status-informers.md

```diff
@@ -0,0 +1,36 @@
+# Conditional Status Informers
+
+Currently, KOTS does not support having conditional application status informers.
+This proposal makes it possible to have a status informer disappear/appear if a specific config option is set to a certain value.
+
+## Goals
+
+- Add the ability to exclude specific status informers based on the applications's config values
+
+## Non Goals
+
+- Make the application status informers support additional kinds/objects.
+
+## Background
+
+For example, many vendors have an embedded vs. external database option.
+They have a status informer set up on the database object, but this only works when using the embedded option.
+Many of them have requested the ability to have conditional status informers so that the status informer can be excluded when selecting the external db option.
+
+## High-Level Design
+
+Add the ability to render/template the `statusInformers` array in the kots application spec (`kind: Application`).
+This way, config template functions (e.g. ConfigOptionEquals) can be used to decide whether or not an informer should be included/applied.
+
+## Detailed Design
+
+## Notes:
+- Getting the list of status informers happens during the app deploy loop in TypeScript.
+- The list of informers is then sent through a socket to the operator to be applied.
+
+1- To add the ability to exclude specific status informers, the `statusInformers` array will be rendered using the `RenderFile` function from kots ffi before it's sent to the operator (the `RenderFile` function can take any string as input to render).
```

Comment:
Is *only* the `statusInformers` array going to be rendered, or will the whole file be rendered?

Reply:
Rendering the whole file would make sense from a user mental-model perspective, but probably has complications, like some content from the file is used prior to having config available. e.g. if the application icon is conditional on config, we wouldn't have that config when first rendering it.

Reply:
Yes, in this proposal, only the `statusInformers` array is going to be rendered. I've discussed this from a technical perspective with @divolgin, and this seems to be the preferable approach because:

1- There are some fields that we shouldn't support rendering/templating for in the application spec (similar to what you pointed above. @divolgin might have a better idea of what those fields are).
2- `kotskinds` objects are usually only rendered on the fly (except for the config) and the rendered version is not stored in the database or s3 (because `kotskinds` are not part of the deployable manifests). This way, the spec can be rendered when needed without modifying the original data.

@divolgin can give his input on this one too.

Reply:
Thanks, that makes sense.